crowbar_framework/lib/crowbar/checks/network.rb

```diff
@@ -0,0 +1,173 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "resolv"
+require "ipaddr"
+require "socket"
+require "logger"
+
+module Crowbar
+  module Checks
+    class Network
+      attr_accessor :logger
+
+      class << self
+        def ip_addrs(fqdn, version = :ipv4)
+          ip_addrs = []
+
+          Resolv.getaddresses(fqdn).each do |address|
+            ip_addr = IPAddr.new(address)
+            if version == :ipv6
+              ip_addrs.push address if ip_addr.ipv6?
+            else
+              ip_addrs.push address if ip_addr.ipv4?
+            end
+          end
+
+          ip_addrs
+        end
+      end
+
+      def initialize(logger = ::Logger.new(STDOUT))
+        self.logger = logger
+      end
+
+      def check!
+        [
+          :fqdn_detected?,
+          :ip_resolved?,
+          :loopback_resolved?,
+          :ip_configured?,
+          :network_valid?,
+          :firewall_disabled?,
+          :ping_succeeds?
+        ].each do |check|
+          return false unless send(check)
+        end
+
+        true
+      end
+
+      def fqdn_detected?
+        if hostname.blank? || domain.blank? || fqdn.blank?
+          logger.error("Unable to detect fully-qualified hostname.")
+          return false
+        end
+
+        true
+      end
+
+      def ip_resolved?
+        if ipv4_addrs.empty? && ipv6_addrs.empty?
+          logger.error("Could not resolve #{fqdn} to an IPv4 or IPv6 address.")
+          return false
+        end
+
+        true
+      end
+
+      def loopback_resolved?
+        if ipv4_addrs.find { |e| /^127/ =~ e } || ipv6_addrs.find { |e| /^::[0-9]$/ =~ e }
+          logger.error("#{fqdn} resolves to a loopback address.")
```

Comment:
hmm all the loggers would now return to STDOUT. shouldn't the log happen in the script that uses this lib?